foreshadow/console.py

```diff
@@ -0,0 +1,244 @@
+    parser.add_argument(
+        "--time",
+        default=10,
+        type=int,
+        help="Time limit in minutes to apply to model"
+        "parameter search. (Default 10)",
+    )
+    parser.add_argument(
+        "--x_config",
+        default=None,
+        type=str,
+        help="Path to JSON configuration file for X Preprocessor",
+    )
+    parser.add_argument(
+        "--y_config",
+        default=None,
+        type=str,
+        help="Path to JSON configuration file for y Preprocessor",
+    )
+    cargs = parser.parse_args(args)
+
+    if cargs.level == 3 and cargs.method is not None:
+        warnings.warn(
+            "WARNING: Level 3 model search enabled. Method will be ignored."
+        )
+
+    if cargs.level != 3 and cargs.time != 10:
+        warnings.warn(
+            "WARNING: Time parameter not applicable "
+            "to feature engineering. Must be in level 3."
+        )
+
+    try:
+        df = pd.read_csv(cargs.data)
+    except Exception:
+        raise ValueError(
+            "Failed to load file. Please verify it exists and is a valid CSV."
+        )
+
+    try:
+        X_df = df.drop(columns=cargs.target)
+        y_df = df[[cargs.target]]
+    except Exception:
+        raise ValueError("Invalid target variable")
+
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_df, y_df, test_size=0.2
+    )
+
+    if cargs.level == 1:
+        # Default everything with basic estimator
+        fs = Foreshadow(estimator=get_method(cargs.method, y_train))
+
+    elif cargs.level == 2:
+        # Parameter search on all matched intents
+
+        if cargs.x_config is not None:
+            try:
+                with open(cargs.x_config, "r") as f:
+                    X_search = Preprocessor(from_json=json.load(f))
+            except Exception:
+                raise ValueError(
+                    "Could not read X config file {}".format(cargs.x_config)
+                )
+            print("Reading config for X Preprocessor")
+        else:
+            X_search = search_intents(X_train)
+            print("Searching over valid intent space for X data")
+
+        if cargs.y_config is not None:
+            try:
+                with open(cargs.y_config, "r") as f:
+                    y_search = Preprocessor(from_json=json.load(f))
+            except Exception:
+                raise ValueError(
+                    "Could not read y config file {}".format(cargs.y_config)
+                )
+            print("Reading config for y Preprocessor")
+        else:
+            y_search = search_intents(y_train)
+            print("Searching over valid intent space for y data")
+
+        # If level 3 also do model parameter search with AutoEstimator
+        # Input time limit into Foreshadow to be passed into AutoEstimator
+
+        fs = Foreshadow(
+            X_preprocessor=X_search,
+            y_preprocessor=y_search,
+            estimator=get_method(cargs.method, X_train),
```

Comment:
X_train is still used here. We need to use y_train I believe. Not sure how the test passed.